File: February2022/Kotlin/MaximumDepthofBinaryTree.kt
//Author: Shobhit Behl (LeetCode:shobhitbruh)
class Solution {
    fun maxDepth(root: TreeNode?): Int {
            return if (root == null) {
                0
            } else Math.max(maxDepth(root.left), maxDepth(root.right)) + 1
        }
}

File: February2022/Python/MaximumDepthofBinaryTree.py
# TC : O(N)
# SC : O(N)
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        left = 1+self.maxDepth(root.left)
        right = 1+self.maxDepth(root.right)
        return max(left, right)
